src/main.rs

```diff
@@ -1,10 +1,154 @@
-// extern crate clap;
+#[macro_use]
+extern crate clap;
 
 mod client;
 mod model;
 mod server;
 mod view;
 
+use clap::AppSettings;
+
 fn main() {
+    // initialize the CLI
+    let battleship = clap_app!(battleship =>
+        (version: crate_version!())
+        (author: crate_authors!())
+        (about: "Two player battleship game")
+        (@subcommand server =>
+            (about: "Server instance for the game")
+            (version: crate_version!())
+            (author: crate_authors!())
+            (@arg port: +required +takes_value "listening port")
+            (@arg name: +required +takes_value "Name of player")
+            (@arg size: -s --size +takes_value
+                "set N as board dimension => N x N [not yet implemented]"
+            )
+            (@arg ships: --ships +takes_value "load ship configuration [not yet implemented]")
+        )
+        (@subcommand client =>
+            (about: "Client instance for the game")
+            (version: crate_version!())
+            (author: crate_authors!())
+            (@arg ip: +required +takes_value "Connect to address")
+            (@arg port: +required +takes_value "Connect to port")
+            (@arg name: +required +takes_value "Name of player")
+        )
+        (@subcommand single =>
+            (about: "Play against the computer")
+            (version: crate_version!())
+            (author: crate_authors!())
+            (@arg name: +required +takes_value "Name of player")
+        )
+    )
+        .setting(AppSettings::SubcommandRequired)
+        .get_matches();
+
     println!("Welcome to a round of 'battleship'");
+
+    match battleship.subcommand() {
+        ("server", Some(server_args)) => {
+            // required arguments
+            let port = validate_port(server_args.value_of("port").unwrap());
+            let name = server_args.value_of("name").unwrap();
+
+            // optional arguments
+            let mut size = 10;
```

Comment:
It's not hard-coded - just the default value if no custom value is given (see lines below)..